balboa-agent/src/main/scala/com/socrata/balboa/agent/BalboaAgent.scala

```diff
@@ -52,15 +40,29 @@ object BalboaAgent extends App with Config with LazyLogging {
   logger info s"Starting Balboa Agent.  Consuming metrics from ${dataDir.getAbsolutePath}.  " +
     s"AMQ Server: $activemqServer, AMQ Queue: $activemqQueue $additionalInfo".trim
 
-  val mc = (activemqUser, activemqPassword) match {
-    case (Some(user), Some(password)) => new MetricConsumer(
-      dataDir, MetricJmsQueue.getInstance(user, password, activemqServer, activemqQueue))
-    case _ => new MetricConsumer(dataDir, MetricJmsQueue.getInstance(activemqServer, activemqQueue))
+  logger.debug("Attempting to create Activemq Connection")
+  val amqConnection = (activemqUser, activemqPassword) match {
+    case (Some(user), Some(password)) => ActiveMQConnection.makeConnection(user, password, activemqServer)
+    case _ => ActiveMQConnection.makeConnection(activemqServer)
   }
-  val future: ScheduledFuture[_] = scheduler.scheduleWithFixedDelay(mc, INITIAL_DELAY, period, TimeUnit.MILLISECONDS)
+  logger.debug(s"Created Activemq Connection $amqConnection")
+
+  // TODO (Pre req: Java 8) Use Java Duration class.
+  val future: ScheduledFuture[_] = scheduler.scheduleWithFixedDelay(new Runnable {
+    override def run(): Unit = {
+      // Wrapping the Metric Consumer inside an anonymous inner function makes the metric consumer more ephemeral
```

Comment:
I found that having an internal buffer that flushes based off of size could result in metric loss if the buffer fills to anything buts its flushing capacity and then gets GCed.  In order to handle that case and preserve a simple abstraction using a well known abstraction I had Metric Consumer implement AutoCloseable.  When MetricConsumer.close is invoked the buffer attempts to flush.  So here I am wrapping MetricConsumer run and close in one Runnable anonymous class to ensure that all the metric messages are sent to AMQ.